--- Assistants/avs-device-sdk/SampleApp/src/CMakeLists.txt	2018-03-20 20:26:45.000000000 +0900
+++ ./CMakeLists.txt	2018-03-17 20:14:43.000000000 +0900
@@ -9,7 +9,9 @@
     UIManager.cpp
     UserInputManager.cpp
     SampleApplication.cpp
-    main.cpp)
+    main.cpp
+    PosixQueueManager.cpp
+    )
 
 IF (HAS_EXTERNAL_MEDIA_PLAYER_ADAPTERS)
     file(GLOB_RECURSE SRC_FILE ${CMAKE_CURRENT_SOURCE_DIR}/ExternalMediaAdapterRegistration/*.cpp)
