diff --git a/SampleApp/include/SampleApp/SampleApplication.h b/SampleApp/include/SampleApp/SampleApplication.h
index 74c6a4498..9adcae1a0 100644
--- a/SampleApp/include/SampleApp/SampleApplication.h
+++ b/SampleApp/include/SampleApp/SampleApplication.h
@@ -27,6 +27,8 @@
 #include "UserInputManager.h"
 #include "SampleApp/GuiRenderer.h"
 
+#include "PosixQueueManager.h"
+
 #ifdef KWD
 #include <KWD/AbstractKeywordDetector.h>
 #endif
@@ -221,6 +223,8 @@ private:
     /// The singleton map from @c playerId to @c ExternalMediaAdapter creation functions.
     static capabilityAgents::externalMediaPlayer::ExternalMediaPlayer::AdapterCreationMap m_adapterToCreateFuncMap;
 
+    std::shared_ptr<PosixQueueManager> m_posixQueueManager;
+    
 #ifdef KWD
     /// The Wakeword Detector which can wake up the client using audio input.
     std::unique_ptr<kwd::AbstractKeywordDetector> m_keywordDetector;
diff --git a/SampleApp/src/CMakeLists.txt b/SampleApp/src/CMakeLists.txt
index c2cb865c9..ddea284a3 100644
--- a/SampleApp/src/CMakeLists.txt
+++ b/SampleApp/src/CMakeLists.txt
@@ -9,7 +9,9 @@ list(APPEND SampleApp_SOURCES
     UIManager.cpp
     UserInputManager.cpp
     SampleApplication.cpp
-    main.cpp)
+    main.cpp
+    PosixQueueManager.cpp
+    )
 
 if (PORTAUDIO)
     list(APPEND SampleApp_SOURCES PortAudioMicrophoneWrapper.cpp)
diff --git a/SampleApp/src/SampleApplication.cpp b/SampleApp/src/SampleApplication.cpp
index 4fba8ed39..b9cb7bb22 100644
--- a/SampleApp/src/SampleApplication.cpp
+++ b/SampleApp/src/SampleApplication.cpp
@@ -217,6 +217,10 @@ SampleApplication::MediaPlayerRegistration::MediaPlayerRegistration(
 }
 
 SampleAppReturnCode SampleApplication::run() {
+
+    avsCommon::utils::threading::Executor executor;
+    executor.submit([this]() { m_posixQueueManager->run(); });
+
     return m_userInputManager->run();
 }
 
@@ -509,6 +513,8 @@ bool SampleApplication::initialize(
         return false;
     }
 
+ 
+        
     authDelegate->addAuthObserver(userInterfaceManager);
     m_capabilitiesDelegate->addCapabilitiesObserver(userInterfaceManager);
 
@@ -583,7 +589,6 @@ bool SampleApplication::initialize(
     client->addSpeakerManagerObserver(userInterfaceManager);
 
     client->addNotificationsObserver(userInterfaceManager);
-
     /*
      * Add GUI Renderer as an observer if display cards are supported.
      */
@@ -724,7 +729,10 @@ bool SampleApplication::initialize(
         ACSDK_CRITICAL(LX("Failed to create UserInputManager!"));
         return false;
     }
-
+    m_posixQueueManager = alexaClientSDK::sampleApp::PosixQueueManager::create(m_interactionManager);
+    client->addAlexaDialogStateObserver(m_posixQueueManager);
+ 
+        
     authDelegate->addAuthObserver(m_userInputManager);
     client->getRegistrationManager()->addObserver(m_userInputManager);
     m_capabilitiesDelegate->addCapabilitiesObserver(m_userInputManager);
diff --git a/SampleApp/src/UserInputManager.cpp b/SampleApp/src/UserInputManager.cpp
index c20011e57..0fb73cc1f 100644
--- a/SampleApp/src/UserInputManager.cpp
+++ b/SampleApp/src/UserInputManager.cpp
@@ -20,7 +20,7 @@
 #include <AVSCommon/Utils/String/StringUtils.h>
 #include "SampleApp/UserInputManager.h"
 #include "SampleApp/ConsolePrinter.h"
-
+#include "unistd.h"
 namespace alexaClientSDK {
 namespace sampleApp {
 
@@ -270,7 +270,8 @@ SampleAppReturnCode UserInputManager::run() {
             }
 #endif
         } else {
-            m_interactionManager->errorValue();
+           //  m_interactionManager->errorValue();
+            usleep(500 * 1000);
         }
     }
     if (!userTriggeredLogout && m_restart) {
