--- SampleApp/include/SampleApp/SampleApplication.h	2018-03-21 10:51:37.000000000 +0900
+++ ./SampleApplication.h	2018-03-17 20:30:45.538633859 +0900
@@ -21,6 +21,7 @@
 
 #include "ConsolePrinter.h"
 #include "UserInputManager.h"
+#include "PosixQueueManager.h"
 
 #ifdef KWD
 #include <KWD/AbstractKeywordDetector.h>
@@ -158,6 +159,8 @@
     /// The singleton map from @c playerId to @c ExternalMediaAdapter creation functions.
     static capabilityAgents::externalMediaPlayer::ExternalMediaPlayer::AdapterCreationMap m_adapterToCreateFuncMap;
 
+    std::shared_ptr<PosixQueueManager> m_posixQueueManager;
+
 #ifdef KWD
     /// The Wakeword Detector which can wake up the client using audio input.
     std::unique_ptr<kwd::AbstractKeywordDetector> m_keywordDetector;
